ledger/participant-integration-api/src/main/scala/platform/indexer/MeteringAggregator.scala

```diff
@@ -0,0 +1,180 @@
+// Copyright (c) 2022 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.platform.indexer
+
+import com.daml.ledger.offset.Offset
+import com.daml.ledger.participant.state.index.v2.MeteringStore
+import com.daml.ledger.participant.state.index.v2.MeteringStore.ParticipantMetering
+import com.daml.ledger.resources.ResourceOwner
+import com.daml.lf.data.Time.Timestamp
+import com.daml.logging.{ContextualizedLogger, LoggingContext}
+import com.daml.metrics.Metrics
+import com.daml.platform.indexer.MeteringAggregator.{toOffsetDateTime, toTimestamp}
+import com.daml.platform.store.appendonlydao.{DbDispatcher, SqlExecutor}
+import com.daml.platform.store.backend.ParameterStorageBackend.LedgerMeteringEnd
+import com.daml.platform.store.backend.{MeteringStorageWriteBackend, ParameterStorageBackend}
+
+import java.sql.Connection
+import java.time.temporal.ChronoUnit
+import java.time.{OffsetDateTime, ZoneOffset}
+import java.util.{Timer, TimerTask}
+import scala.concurrent.duration._
+import scala.concurrent.{Await, ExecutionContext, Future}
+import scala.util.{Failure, Success}
+
+object MeteringAggregator {
+
+  class Owner(
+      meteringStore: MeteringStorageWriteBackend,
+      parameterStore: ParameterStorageBackend,
+      metrics: Metrics,
+      period: FiniteDuration = 6.minutes,
+  ) {
+
+    private[platform] def apply(
+        dispatcher: DbDispatcher
+    )(implicit loggingContext: LoggingContext): ResourceOwner[Unit] = {
+      val aggregator = new MeteringAggregator(meteringStore, parameterStore, metrics)
+      Await.ready(aggregator.initialize(dispatcher), 60.seconds)
+      ResourceOwner.forTimer(() => new Timer()).map { timer =>
+        timer.scheduleAtFixedRate(
+          new TimerTask {
+            override def run(): Unit = {
+              Await.ready(aggregator.run(dispatcher), 60.minutes)
+              ()
+            }
+          },
+          period.toMillis,
+          period.toMillis,
+        )
+      }
+    }
+  }
+
+  private def toTimestamp(dateTime: OffsetDateTime): Timestamp =
+    Timestamp.assertFromInstant(dateTime.toInstant)
+  private def toOffsetDateTime(timestamp: Timestamp): OffsetDateTime =
+    OffsetDateTime.ofInstant(timestamp.toInstant, ZoneOffset.UTC)
+
+}
+
+class MeteringAggregator(
+    meteringStore: MeteringStorageWriteBackend,
+    parameterStore: ParameterStorageBackend,
+    metrics: Metrics,
+    clock: () => Timestamp = () => Timestamp.now(),
+)(implicit loggingContext: LoggingContext) {
+
+  private implicit val ec: ExecutionContext = ExecutionContext.global
```

Comment:
please do not use ExecutionContexts directly if possible: this should be either an execution context passed down from above, or parasitic (I think you can go with parasitic, since not much this is used anyway (putting it differently: all the work will be done in the context of a database transaction and the thread associated))

Reply:
We really try to avoid using global execution context.

Reply:
Switched to using parasitic.